src/codegen/baseline_jit.cpp

```diff
@@ -870,8 +872,13 @@ Box* JitFragmentWriter::createListHelper(uint64_t num, Box** data) {
 
 Box* JitFragmentWriter::createSetHelper(uint64_t num, Box** data) {
     BoxedSet* set = (BoxedSet*)createSet();
-    for (int i = 0; i < num; ++i)
-        set->s.insert(data[i]);
+    for (int i = 0; i < num; ++i) {
+        auto&& p = set->s.insert(data[i]);
+        if (!p.second /* already exists */) {
+            Py_DECREF(p.first->value);
+            *p.first = data[i];
```

Comment:
I think this should use the original value and decref the duplicate one: `set([1, 1L])` is `1` for CPython and `1L` for us.